core/parentchain/indirect-calls-executor/src/indirect_calls_executor.rs

```diff
@@ -167,23 +186,94 @@ where
 			})
 			.unwrap_or(false)
 	}
+
+	fn is_ack_game_function(&self, function: &[u8; 2]) -> bool {
+		self.node_meta_data_provider
+			.get_from_metadata(|meta_data| {
+				let call = match meta_data.ack_game_call_indexes() {
+					Ok(c) => c,
+					Err(e) => {
+						error!("Failed to get the indexes for the ack_game call from the metadata: {:?}", e);
+						return false
+					},
+				};
+				function == &call
+			})
+			.unwrap_or(false)
+	}
+
+	fn is_finish_game_function(&self, function: &[u8; 2]) -> bool {
+		self.node_meta_data_provider
+			.get_from_metadata(|meta_data| {
+				let call = match meta_data.finish_game_call_indexes() {
+					Ok(c) => c,
+					Err(e) => {
+						error!("Failed to get the indexes for the finish_game call from the metadata: {:?}", e);
+						return false
+					},
+				};
+				function == &call
+			})
+			.unwrap_or(false)
+	}
+
+	fn handle_ack_game_xt<ParentchainBlock>(
+		&self,
+		xt: &ParentchainUncheckedExtrinsic<AckGameFn>,
+		block: &ParentchainBlock,
+	) -> Result<()>
+	where
+		ParentchainBlock: ParentchainBlockTrait<Hash = H256>,
+	{
+		let (_call, games, shard) = &xt.function;
+
+		info!("found {:?} games", games.len());
+
+		// TODO: We should actually submit a call here, not execute directly.
+		for game in games {
+			self.stf_game_executor.new_game(*game, shard, block)?;
```

Comment:
It should work like that (still) but it's not very clean and future-proof to access the state directly. I'd recommend switching to call submission to the pool (see https://github.com/ajuna-network/worker/blob/bh/ajuna-squashed/core/parentchain/indirect-calls-executor/src/indirect_calls_executor.rs#L111-L135) such that the extrinsic gets executed as a batch with the next block.

Same for the `handle_finish_game_xt` below.